src/SMU-Keithley_2400/main.py

```diff
@@ -25,72 +25,83 @@
 
 # SweepMe! device class
 # Type: SMU
 # Device: Keithley 2400
 
 
-from EmptyDeviceClass import EmptyDevice
+from pysweepme.EmptyDeviceClass import EmptyDevice
 import numpy as np
 import time
 
 
 class Device(EmptyDevice):
 
     def __init__(self):
-        
+
         EmptyDevice.__init__(self)
-        
-        self.shortname = "Keithley2400"
-        
-        self.variables = ["Voltage", "Current"]
-        self.units = ["V", "A"]
-        self.plottype = [True, True]  # True to plot data
-        self.savetype = [True, True]  # True to save data
 
+        self.shortname = "Keithley2400"
         self.port_manager = True
         self.port_types = ["COM", "GPIB"]
-        
+
         self.port_properties = {
             "timeout": 10,
             "EOL": "\r",
             "baudrate": 9600,
         }
-                                 
+
         self.commands = {
             "Voltage in V": "VOLT",
             "Current in A": "CURR",
             "Voltage [V]": "VOLT",  # deprecated, remains for compatibility reasons
             "Current [A]": "CURR",  # deprecated, remains for compatibility reasons
         }
-                        
-        self.outpon = False 
-                                 
+
+        self.outpon = False
+
     def set_GUIparameter(self):
-        
+
         gui_parameter = {
             "SweepMode": ["Voltage in V", "Current in A"],
             "RouteOut": ["Front", "Rear"],
-            "Speed": ["Fast", "Medium", "Slow"],
-            "Range": ["Auto",
-                      "10 nA",
-                      "100 nA",
-                      "1 µA",
-                      "10 µA",
-                      "100 µA",
-                      "1 mA",
-                      "10 mA",
-                      "100 mA",
-                      "1 A", ],
+            "Speed": ["Fast", "Medium", "Slow", "Very fast"],
+            "Range": [
+                "Auto",
+                "10 nA",
+                "100 nA",
+                "1 µA",
+                "10 µA",
+                "100 µA",
+                "1 mA",
+                "10 mA",
+                "100 mA",
+                "1 A",
+            ],
             "Compliance": 100e-6,
             "Average": 1,
             "4wire": False,
+
+            "ListSweepCheck": True,
+            "ListSweepType": ["Sweep", "Custom"],
+            "ListSweepStart": 0.5,
+            "ListSweepEnd": 1.0,
+            "ListSweepStepPointsType": ["Step width:", "Points (lin.):", "Points (log.):"],
+            "ListSweepStepPointsValue": 0.1,
+            "ListSweepCustomValues": "",
+            "ListSweepDual": False,
+            "ListSweepHoldtime": 0.1,
+            "ListSweepDelaytime": 0.1,
         }
-                        
+
         return gui_parameter
-                                 
+
     def get_GUIparameter(self, parameter={}):
+
+        self.sweepvalue = parameter["SweepValue"]
```

Comment:
a default for SweepValue should be added in set_GUIparameter, otherwise the driver fails in pysweepme with
`KeyError: 'SweepValue'`

Reply:
thanks, added it now